lib/archethic/beacon_chain/network_coordinates.ex

```diff
@@ -0,0 +1,193 @@
+defmodule Archethic.BeaconChain.NetworkCoordinates do
+  @moduledoc """
+  Network coordinates is a way to map latency between nodes and used to determine the closest nodes
+  """
+
+  @digits ["F", "E", "D", "C", "B", "A", "9", "8", "7", "6", "5", "4", "3", "2", "1", "0"]
+
+  @type latency_patch :: {String.t(), String.t()}
+
+  @doc """
+  Compute the network patch based on the matrix latencies
+
+  The matrix must be an Hollow matrix, symmetric and all the latencies must be positive
+
+  In physics, the center of mass for a distribution of masses in the space is the unique point where
+  the weighted relative position of the distributed mass sums to zero. Here, the sum of vectors to
+  all nodes from the center of mass is zero (All the nodes being the unit of mass)
+
+  The distance between center of mass and each node is then reduced to this formula using law of cosine
+  and center of mass:
+  `$$ dic^2 =  {1\over 2} \sum_{j=1}^{n} dij^2 - {1\over 2}\sum_{j=2}^{n}\sum_{k=1}^{j-1} djk^2 $$`
+
+  Then using the laws of Cosine, we can compute a gram matrix which would help to plot the nodes in a 2D plan:
+  `$$ gij =  {1\over 2} (dic^2 + djc^2 - dij^2) $$`
```

Comment:
I can't display correctly the formula are you using some special extension ?

Reply:
I used Markdown editor like: https://stackedit.io/ with Mathematics plugin

Reply:
so in  here a little typo the correct formula is 1/n not 1/2 (in the implementation it's correct, typo only in doc)